x/liquidstakeibc/keeper/abci.go

```diff
@@ -82,11 +85,27 @@ func (k *Keeper) DoDelegate(ctx sdk.Context, hc *types.HostChain) {
 	}
 
 	// if everything went well, update the deposit states and set the sequence id
+	lastEpoch := int64(0) // highest epoch among deposits, used for event emission
 	for _, deposit := range deposits {
 		deposit.IbcSequenceId = sequenceID
 		deposit.State = types.Deposit_DEPOSIT_DELEGATING
 		k.SetDeposit(ctx, deposit)
+
```

Comment:
why not log individual deposits, just incase there was a failure in previous one, so we can know.

Reply:
because we generate messages based on the total amount, not individual deposits, but we can add an additional event  just for individual deposits as well

Reply:
we should add it even if it seems redundant because all this data we are adding is towards building the backend for tracking things. 
we should let the backend decide which ones should be used.